self_test/SelfTest.ts

```diff
@@ -0,0 +1,120 @@
+
+import { Injector } from 'reduct'
+import Config from '../src/services/Config'
+import { create as createLogger } from '../src/common/log'
+const ilpFetch = require('ilp-fetch')
+const log = createLogger('SelfTest')
+const WebSocket = require('ws')
+const manifestJson = require('./self-test-manifest.json')
+const axios = require('axios')
+const crypto = require('crypto')
+export default class SelfTest {
+  private config: Config
+  constructor (deps: Injector) {
+    this.config = deps(Config)
+  }
+
+  start () {
+    if (!this.config.disableSelfTest) {
+      this.run()
+      .catch(err => log.error(err))
+    } else {
+      log.debug('Skipping self test')
+    }
+  }
+
+  async run () {
+    const duration = 300
+    const host = this.config.publicUri
+    try {
+      const randomName = crypto.randomBytes(20).toString('hex')
+      manifestJson['manifest']['name'] = randomName
+      log.debug('manifestJson', manifestJson)
+      let response = await ilpFetch(`${host}/pods?duration=${duration}`, {
+        headers: {
+          Accept: `application/codius-v1+json`,
+          'Content-Type': 'application/json'
+        },
+        maxPrice: (this.config.hostCostPerMonth * 1000000).toString(),
+        method: 'POST',
+        body: JSON.stringify(manifestJson),
+        timeout: 70000 // 1m10s
+      })
+      log.debug('ilpFetch Resp', response)
+      if (this.checkStatus(response)) {
+        // Maybe check status in 30 seconds interval twice.
+        response = await response.json()
+        const url = new URL(this.config.publicUri)
+        await new Promise((resolve, reject) => {
+          setTimeout(() => {
+            resolve()
+          }, 10000)
+        })
+        const webSocketPromise = new Promise(resolve => {
+          const ws = new WebSocket(`wss://${response.manifestHash}.${url.host}/websockets`)
+          ws.on('open', () => {
+            log.debug('Web sockets are enabled')
+          })
+
+          ws.on('message', (message: string) => {
+            let finalMessage = JSON.parse(message)
+            if (finalMessage.websocketEnabled) {
+              resolve()
+            }
+          })
+
+          ws.on('error', (err: any) => {
```

Comment:
An error event may be fired after this promise is resolved if the websocket is still open, and because we're calling process.exit it could crash the process several minutes later. We could reject the promise here instead of running process.exit, and/or we could unbind all the event listeners on the websocket when we resolve